### DISCUSSION TITLE
open action creator props cannot have a property named type

### DISCUSSION THREAD

**Discussion by tirelyl** [ngrx/platform#3589] [Q&A] [answered]
I have a feature reducer state witch the interface already contains the property named `type`. when i create action by use the state, i got the an error below:
<img width="761" alt="image" src="https://user-images.githubusercontent.com/20921324/192189258-a9ed7597-5632-466f-837e-78e6c3427008.png">
<img width="626" alt="image" src="https://user-images.githubusercontent.com/20921324/192189915-099eff1a-9540-4b20-93b6-6a076fab5351.png">

I know this is because the action's state is generate by ts union type `{ type: string } & MyCustomState`, already contains the type attribute. So, I wonder if there is a way to bypass this error，such as rename the `type`, or other more efficient methods.

I just started learning ngrx recently, I hope good-hearted people can help to answer

**Comment by timdeschryver** [comment, marked as answer]
Hi @tirelyl this is indeed something that needs to work this way. An action is required to have a `type`, based on this property the reducer(s) and effect(s) know what to do with it.
Because it's the core of NgRx (the Redux pattern), you can't change its name. 

To solve the problem, you'll have to rename the `type` property on your end, or wrap it in an object.
For example

```
// "bad"
const myAction = createAction('[Source] Event', props<User>())
// good
const myAction = createAction('[Source] Event', props<{ user: User }>())
```

For more info see https://ngrx.io/guide/eslint-plugin/rules/prefer-inline-action-props

**Reply by tirelyl** [reply]
thanks for your reply! I choose to rename `type` to `wsType` in the end. 

I think it would be more elegant to have an api that changes the `type` name. But that's okay, it looks like it's just a small problem
